### PR TITLE
Add script for calculating available development weeks

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Description

This PR adds a script which can be used to (imperfectly) capture the available number of weeks of development time for the Openverse maintainers for the next year. This is based off work by @sarayourfriend done here: https://gist.github.com/sarayourfriend/5f16224fe4897926209fa1a3e06905fe.

For the current configuration, this outputs **~272 available development weeks**.

## Testing Instructions

1. `cd utilities/project_planning`
2. `pipenv run calculate_available_weeks.py`

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: utilities/project_planning/calculate_available_weeks.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+1. The number of maintainers stays the same
+2. We cannot anticipate availability changes like parental leave, part-time moves, etc.
+3. 5 weeks of holiday + sick leave with stipulation that this is just an estimate!!!!!
+4. Not all conferences are accounted for
+"""
+
+programmers_with_reduced_availability = {
+    # _ is part-time (safer to assume that isn't going to change)
+    "_": (52 / 2) * 0.6,
```
Comment (sarayourfriend) [inline utilities/project_planning/calculate_available_weeks.py:14]:
```suggestion
    "_": 52 * 0.6,
```

Let's assume that `_` is full time for the full year, not just half.

File: utilities/project_planning/calculate_available_weeks.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+conference_weeks = {
+    "wceu": 3,
+    "wcus": 2,
+    "wccs": 2,
+    "wcasia": 2,
+    "ccglobalsummit": 1,
+    "data": 1,
+    "pydata": 0.5,
+    "pycascades": 1,
+    "widsps": 0.25,
```
Comment (sarayourfriend) [inline utilities/project_planning/calculate_available_weeks.py:50]: Out of curiosity, what meetup is this?

File: utilities/project_planning/calculate_available_weeks.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        total_openverse_maintainers - len(programmers_with_reduced_availability)
+    )
+}
+
+programmer_availability_ignoring_regular_leave = (
+    programmers_with_reduced_availability | programmers_with_regular_availability
+)
+
+# These are just vague estimates, obviously some folks take more or less, etc.
+holiday_weeks = 4
```
Comment (sarayourfriend) [inline utilities/project_planning/calculate_available_weeks.py:35]: @AetherUnbound are you able to see the sponsored contributors average AFK for this year and calculate based on that? I'm worried 4 might be too low :thinking: If the average from this year is lower though then let's keep at 4 to assume the sponsored contributors will (correctly :slightly_smiling_face:) take the leave they're expected to.

**Review by sarayourfriend — APPROVED** [review]
LGTM! Non-blocking suggestions.

File: utilities/project_planning/calculate_available_weeks.py (hunk 1 above)
Comment (AetherUnbound) [reply utilities/project_planning/calculate_available_weeks.py:14]: Full time or part time?

File: utilities/project_planning/calculate_available_weeks.py (hunk 2 above)
Comment (AetherUnbound) [reply utilities/project_planning/calculate_available_weeks.py:50]: [Women in Data Science: Puget Sound!](https://www.widspugetsound.org/)

File: utilities/project_planning/calculate_available_weeks.py (hunk 3 above)
Comment (AetherUnbound) [reply utilities/project_planning/calculate_available_weeks.py:35]: I think I can, I'll take a look and adjust if it seems inaccurate!

Comment (AetherUnbound) [reply utilities/project_planning/calculate_available_weeks.py:35]: Update: this number is accurate.

File: utilities/project_planning/calculate_available_weeks.py (hunk 1 above)
Comment (sarayourfriend) [reply utilities/project_planning/calculate_available_weeks.py]: Part time, sorry.

**Review by dhruvkb — APPROVED** [review]
LGTM! It's very thorough and even captures cases I would not have considered like "Not all team members attend division meetups".

Should we consider @fcoveram's availability as a fraction too, considering his role increasingly shifting more towards WordPress.org overall, away from Openverse specifically? Also separating the design weeks + dev weeks might be further optimisation.

**Comment by fcoveram** [comment]
Splitting design and dev sounds good. My contribution to Openverse and other WordPress projects swings during the year as per other designers' availability. So I'm okay with whatever decision you all think is correct.

**Comment by AetherUnbound** [comment]
I'll merge this as-is for now, since I've already used the numerical output from this in the next step of the planning 😅 We can definitely (and will need to) re-evaluate it next year to see if we felt it was accurate!

**Pull request merged**